### PR TITLE
Do not display new products in the BO listing until they are saved

### PR DESCRIPTION
**Pull request by jocel1**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | New products are not visible in the BO until they are saved |
| Type? | improvement |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? |  |
| How to test? | In the BO, when clicking on Add Product, it doesn't appear anymore in the listing until the product is saved. The title placeholder for the product should not be set by default |

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 1]
```diff
@@ -241,6 +241,8 @@ public function listAction(Request $request, $limit = 10, $offset = 0, $orderBy
         $adminProductWrapper = $this->container->get('prestashop.adapter.admin.wrapper.product');
         $totalCount = 0;
 
+        $this->get('prestashop.service.product')->cleanupOldTempProducts();
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: A model class is not a service. More, ``getOldTempProducts`` is a static function :)

How about ``\Product::cleanupOldTempProducts();`` ?

File: src/PrestaShopBundle/Service/ProductService.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Service;
+
+use PrestaShop\PrestaShop\Adapter\Product\ProductDataProvider;
+
+class ProductService {
+    /** @var ProductDataProvider */
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Service/ProductService.php]: I dont understand why you declare a Product model as a service, or maybe the class name is bad ?

**Comment by vincentbz** [comment]
Sorry I cant find any bug here^^
Product is created, I can add images, create combinations, and other actions, before saving and entering the name.
When I create combinations, it just cannot save because the name is empty but it doesn't delete the combinations. 
When the save is done, the product URL contains the product name as it should be. Friendly URL is OK.
Actions (save + things, delete, preview...) work.
Product doesn't appear in the list before the save
And save + duplicate creates the product as it should be.
No pb with categories creation/association

So it's ok for me, big thanks :)
status : 'QA approved'

**Comment by Shudrum** [comment]
Thank you!

**Pull request merged**